Tests/AppTests/Util.swift

```diff
@@ -26,8 +26,13 @@ private var _schemaCreated = false
 
 func setup(_ environment: Environment, resetDb: Bool = true) throws -> Application {
     let app = Application(.testing)
+    let host = try configure(app)
+
+    // Ensure `.testing` refers to "postgres" or "localhost"
+    precondition(host == "localhost" || host == "postgres", ".testing must be a local db, was: \(host)")
```

Comment:
This is very important to note: while testing this PR I discovered that once an `Application` is instantiated via `Application(.testing)`, it is _not_ possible to instantiate another `Application(.staging)` and have it resolve to a different database.

When I did so, the query performance tests kept failing, because even though `app` was instantiated via `Application(.staging)`, the database host the test was connecting to was still `postgres` (i.e. the test db docker container).

That _also_ means that had I for some reason run the query performance tests first, thereby instantiating `Application(.staging)` first and then within the same test run run _another_ test that tries to set up the test db... 💣 I would have torn down staging...

There are now a couple of safe-guards in place against this (this `precondition` and the `RUN_QUERY_PERFORMANCE_TESTS=true` env variable). We can go further and pull this test out into its own target entirely to add another layer of protection.

Reply:
As long as we can never run the main set of tests against the staging database 😅

Reply:
Yeah, I'll make it so it can't be done accidentally :)

I'll tidy this up over the coming days and baseline the relevant queries.

Reply:
It's probably not a good fit for doing at the same time but this discussion also reminded me of #1462 in this area.

Reply:
I suspect this was related to https://github.com/vapor/vapor/issues/2755, which is now fixed. But it’s good to have the safeguard in place anyway.